### ISSUE TITLE
resources need to be able to define parent_context

### ISSUE THREAD

**Issue by trel** [irods/irods_client_zone_management_tool#91]
for compound resources... their children need to be able to define parent_context as either 'cache' or 'archive'.

might need another field in the resource editor pane.

**Comment by connectbo** [comment]
Making sure I understand this, does only compound child resource have `parent_context` field, or all resource children will have but only compound child ones are able to define it?

**Comment by trel** [comment]
they all have it - just that it matters for the compound (the compound resource plugin code pays attention to that field)

**Comment by trel** [comment]
since it's probably just another field, squeezing into 0.1.0

**Comment by connectbo** [comment]
I see. So all child resources have this field and can be modified, so it could be just a textfield? Do we need to validate the input here?

**Comment by trel** [comment]
yes, just a textfield, like the other context...     not for now, no validation.  empty is fine/common.

**Comment by trel** [comment]
ah, it's a bit more complicated it seems... bumping.

**Comment by trel** [comment]
yes, it happens during 'addchildtoresc'... the `ContextString`...

```
$ iadmin h addchildtoresc
 addchildtoresc Parent Child [ContextString] (add child to resource)
Add a child resource to a parent resource.  This creates an 'edge'
between two nodes in a resource tree.
 
Parent is the name of the parent resource.
Child is the name of the child resource.
ContextString is any relevant information that the parent may need in order
  to manage the child.
```

So, perhaps when the user drags a child to a compound resource (in the treeview), we provide a dialog(?) that asks whether this child is to be a 'cache' or an 'archive'.    Kind of clunky... but not sure where else it could fit...
